buildSrc/src/main/groovy/com/cliqz/gradle/TestdroidTask.groovy

```diff
@@ -0,0 +1,202 @@
+package com.cliqz.gradle
+
+import okhttp3.MediaType
+import okhttp3.MultipartBody
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import groovy.json.JsonSlurper
+
+import java.nio.charset.StandardCharsets
+
+class TestDroidTask extends DefaultTask {
+
+    def env = System.getenv()
+    String projectId = env['PROJECT_ID']
+    String apIKey = env['API_KEY']
+    String frameworkId = env['FRAMEWORK_ID']
+    String deviceGroupId = env['DEVICE_GROUP_ID']
+
+
+    final PROJECT_ID = projectId
+    final API_KEY = apIKey
+    final FRAMEWORK_ID = frameworkId
+    final DEVICE_GROUP_ID = deviceGroupId
+
+    final OkHttpClient client = new OkHttpClient()
+    final slurper = new JsonSlurper()
+
+    @TaskAction
+    def connectTestdroid() {
+        // 1. Does the project identified by the id param exists? Decide yourself how to pass this param
+        final url = new URL("https://cloud.bitbar.com/api/me/projects/${PROJECT_ID}")
+        final request = url.openConnection()
+        request.setRequestProperty('Authorization', "Basic ${getBasicAuth(API_KEY)}")
+        if (request.responseCode == HttpURLConnection.HTTP_OK) {
+            logger.info("Project exists")
+        } else {
+            logger.error("No project for the given id: ${PROJECT_ID}")
+            throw new Exception('Error occurred: Specify a project ID that exists')
+        }
+        // 2. Upload the 2 apks and store the ids
+        final apkID = uploadFile("/Users/kiizajosephbazaare/browser-android/app/build/outputs/apk/cliqz/debug/app-cliqz-arm64-v8a-debug.apk")
+        final testsID = uploadFile("/Users/kiizajosephbazaare/browser-android/app/build/outputs/apk/androidTest/cliqz/debug/app-cliqz-debug-androidTest.apk")
+        // 3. Create the testrun and run it
+        final runID = performTestRun(apkID, testsID)
+        // 4. Poll every minute (os so) to check the job finished
+        String state = pollTestRunStatus(runID)
+        int timeTaken = 0
+        while (state!= "FINISHED"){
+            sleep(60000)
+            timeTaken+=1
+            state = pollTestRunStatus(runID)
+        }
+        logger.info("Tests finished in ${timeTaken} minutes")
+        // 5. Download the artifacts and put them in a predefined folder (where Azure/Jenkins can find it)
+        fetchTestRunArtifacts(runID)
+        // 6. log error if any test failed, otherwise just return
+        Float successRatio = errorLogger(runID)
+        if (successRatio == 1){
```

Comment:
Never use == with floating point values. Instead use >= like the following `successRation >= 0.9999`